File: app-backend/src/main/java/dev/andrei/app_backend/dto/review/ReviewDto.java
package dev.andrei.app_backend.dto.review;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

/** A review as shown on a location's public review list. overallScore = mean of attribute scores. */
public record ReviewDto(
        UUID id,
        String reviewerDisplayName,
        String content,
        Instant createdAt,
        double overallScore,
        List<AttributeScoreDto> attributeScores
) {}

File: appfrontend/app/src/main/java/dev/andrei/app_frontend/data/remote/dto/ReviewDto.kt
package dev.andrei.app_frontend.data.remote.dto

// createdAt is an ISO-8601 string (the backend serialises Instant that way).
data class ReviewDto(
    val id: String,
    val reviewerDisplayName: String,
    val content: String,
    val createdAt: String,
    val overallScore: Float,
    val attributeScores: List<AttributeScoreDto>
)
